lib/util/git/index.ts

```diff
@@ -450,13 +450,19 @@ export function getBranchList(): string[] {
 export async function isBranchStale(branchName: string): Promise<boolean> {
   await syncBranch(branchName);
   try {
+    const { currentBranchSha, currentBranch } = config;
     const branches = await git.branch([
       '--remotes',
       '--verbose',
       '--contains',
       config.currentBranchSha,
     ]);
-    return !branches.all.map(localName).includes(branchName);
+    const isStale = !branches.all.map(localName).includes(branchName);
+    logger.debug(
+      { isStale, branches, currentBranch, currentBranchSha },
```

Comment:
This will log the full non-stale branch list once per-branch which calls isStale, right?

i.e. if there are 20 branches and 10 are non-stale then we'd get the same list of 10 branches logged 20 times during a run?

Reply:
yes, but sometimes there must be something wrong. i don't now yet what, so i can reduce to only log `branches.all`  but the `branches.branches` value can be interesting too, when the issue occure. 🤔

Reply:
Maybe `currentbranchSha` is wrong because of a silent git error happened on an other branch before?